app/service/task.go

```diff
@@ -74,6 +74,46 @@ func (t Task) UpdateTask(task entity.Task) error {
 	panic("not implemented")
 }
 
+func (t Task) StartTask(startTaskID oneEntity.ID, userID oneEntity.ID) error {
+	///*
+	//	UI:
+	//		Personal: move startTask to need attention
+	//                  bring down need attention task to upcoming
+	//		Team: move startTask to In Progress
+	//			  move need attention task back to upcoming
+	// */
+	//
+
+	// TODO: a user starts others' task will assign that task to the himself
+	// TODO: need a modal to confirm above
+	activeTeam, err := t.teamRepo.GetActiveTeam(userID)
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	prevNeedAttentionTaskID, err := t.taskRepo.SetNeedAttentionTask(&startTaskID, userID, activeTeam.ID)
+	if err != nil {
+		log.Println(err)
+	}
+
+	err = t.taskRepo.SetTaskStatus(startTaskID, activeTeam.ID, entity.TaskStatusInProgress)
+	if err != nil {
+		log.Println(err)
+	}
+
+	if prevNeedAttentionTaskID == nil {
+		return nil
+	}
+
+	err = t.taskRepo.SetTaskStatus(*prevNeedAttentionTaskID, activeTeam.ID, entity.TaskStatusUpcoming)
+	if err != nil {
+		log.Println(err)
+	}
```

Comment:
```suggestion
	if prevNeedAttentionTaskID != nil {
			err = t.taskRepo.SetTaskStatus(*prevNeedAttentionTaskID, activeTeam.ID, entity.TaskStatusUpcoming)
	        if err != nil {
		        log.Println(err)
		        return err
	        }
	}
	
	return nil
```

Prefer this in case more logics are added in the future